packages/hygraph-app/lib/createRecursiveIntrospectionQuery.ts

```diff
@@ -0,0 +1,13 @@
+export const createRecursiveIntrospectionQuery = (type, depth) => {
```

Comment:
At the places you are using this recursive function, you are still defining a depth. This means you do know the dept it is going to be

Reply:
I don't see the need for a recursive function to generate graphql templates, since this makes the code less readable and nothing is automated by doing it this way. One pro of graphql is an easy to understand syntax, which you lose by doing it this way